src/App/Fossa/Analyze/Log4jReport.hs

```diff
@@ -0,0 +1,299 @@
+          , "-----------------------------------------"
+          , "Lists all projects, reports any log4j dependencies and it's vulnerability type (if any)."
+          , ""
+          , "For more information refer to:"
+          , "- Log4j Vulnerability: https://fossa.com/blog/log4j-log4shell-zero-day-vulnerability-impact-fixes/"
+          , "- Fossa CLI documentation: https://github.com/fossas/fossa-cli/blob/master/README.md"
+          , ""
+          , "To analyze all dependencies (not only log4j), please refer to Fossa CLI."
+          , ""
+          ]
+      )
+  projectReports <- traverse printProjectReport projects
+  logStdout $ renderIt (vsep projectReports)
+
+getVulnerability :: Dependency -> Set Vulnerability
+getVulnerability Dependency{..} =
+  case List.find (\vd -> dependencyName == vdName vd) getVulnerableDeps of
+    Nothing -> Set.empty
+    Just vd -> case (verConstraintToRevision =<< dependencyVersion) of
+      Nothing -> Set.empty
+      Just versionText -> case parse parseSimplifiedVersion "dependency version" versionText of
+        Left _ -> Set.empty
+        Right sv -> Map.findWithDefault Set.empty sv (vdKnownVulnerableVersion vd)
+
+printProjectReport :: (Has Diag.Diagnostics sig m) => ProjectResult -> m (Doc AnsiStyle)
+printProjectReport ProjectResult{..} =
+  pure $
+    vsep
+      [ ""
+      , "Project Path: " <> (pretty . show $ projectResultPath)
+      , pretty $ Text.intercalate "" $ replicate (length $ "Project Path: " <> (show projectResultPath)) "-"
+      , listPretty "Project Manifest Files" (pretty . show <$> projectResultManifestFiles)
+      , listPretty "Direct Inclusions" (map withAnnotation directInclusion)
+      , listPretty "Indirect Inclusions" (map withAnnotation inDirectInclusion)
+      ]
+  where
+    listPretty :: Text -> [Doc AnsiStyle] -> Doc AnsiStyle
+    listPretty header items =
+      if null items
+        then (pretty (header <> ": ")) <> "0 log4j dependencies found!"
+        else vsep $ (pretty header <> ":") : map ("- " <>) items
+
+    directInclusion :: [Log4jVulnerableReportItem]
+    directInclusion = map (toVulnerableReportItem toVulnerableDependencyManifestFiles) onlyDirects
+
+    inDirectInclusion :: [Log4jVulnerableReportItem]
+    inDirectInclusion = map (toVulnerableReportItem toVulnerableDependencyRootDependency) allDeep
+
+    withAnnotation :: Log4jVulnerableReportItem -> Doc AnsiStyle
+    withAnnotation (Log4jVulnerableReportItem name version origin vulTypes) =
+      colorCoded $
+        Text.intercalate
+          " "
+          [ name
+          , formattedVersion
+          , formattedVulnerability
+          , toText . show $ origin
+          ]
+      where
+        colorCoded doc = if Set.null vulTypes then (pretty doc) else (annotateWarn $ pretty doc)
+        formattedVersion = maybe "N/A" ("v" <>) (verConstraintToRevision =<< version)
+        formattedVulnerability =
+          if Set.null vulTypes
+            then inBracket "N/A"
```

Comment:
Todo: Use `Safe`?

Reply:
Definitely use `Safe`, in green if possible